cli/cli-utils/src/getConfig.ts

```diff
@@ -23,6 +26,20 @@ export async function getConfig (
     ignoreNonAuthSettingsFromLocal: opts.ignoreNonAuthSettingsFromLocal,
   })
   config.cliOptions = cliOptions
+  if (config.configDependencies) {
+    const store = await createOrConnectStoreController(config)
+    await installConfigDeps(config.configDependencies, {
+      registries: config.registries,
+      rootDir: config.lockfileDir ?? config.rootProjectManifestDir,
+      store: store.ctrl,
+    })
+  }
+  if (!config.ignorePnpmfile) {
+    config.hooks = requireHooks(config.lockfileDir ?? config.dir, config)
+    if (config.hooks?.updateConfig) {
+      config = config.hooks.updateConfig(config)
```

Comment:
Do we need to check the return value of `config.hooks.updateConfig(config)`? If it is not a valid result, should we give a warning or fallback?

Reply:
ok, done